.github/workflows/aws-deploy.yml

```diff
@@ -18,26 +20,22 @@ jobs:
       - name: Checkout repository
         uses: actions/checkout@v3
 
-      - name: Static Analysis
-        uses: pre-commit/action@v3.0.0
-
       - name: cdk synth
         uses: youyo/aws-cdk-github-actions@v2
         with:
           cdk_subcommand: 'synth'
           actions_comment: false
-          cdk_args: '--context env=dev --debug'
+          cdk_args: '--context env=${{ inputs].CONTEXT }} --debug'
```

Comment:
i think it's a bad idea to put this test here because stuff in here run after a merge.  A `synth` does not need secrets therefore you can and should run it on every PR.